Core/include/Acts/Vertexing/GaussianTrackDensity.ipp

```diff
@@ -78,6 +78,9 @@ void Acts::GaussianTrackDensity<input_track_t>::addTracks(
     const double d0 = boundParams.parameters()[BoundIndices::eBoundLoc0];
     const double z0 = boundParams.parameters()[BoundIndices::eBoundLoc1];
     // Get track covariance
+    if (not boundParams.covariance().has_value()) {
+      continue;
```

Comment:
I would assume this is an error?

Reply:
The problem here is that this function returns `void` and not e.g. a  `Result` type. If we want to signal an error here we can just throw and exception (e.g. by using `.value`)?

Reply:
Do we have a logger here? I think that would help make the error more traceable. Then we could either do the exception or just an `std::terminate`. In a follow-up PR, we could switch the return to `Result<void>`. What do you think?

Reply:
The function returns `void`, so it was trivial to change it to `Result<void>` and use `VertexingError::NoCovariance` to signal the error. This function is called in a single other function which already returns a special value if its computation is invalid, so I did the same thing for the `addTracks`. Let me know if that works!